addons/godot-xr-tools/effects/fade.gd

```diff
@@ -0,0 +1,79 @@
+@tool
+class_name XRToolsFade
+extends Node3D
+
+
+## XR Tools Fade Script
+##
+## This script manages fading the view.
+
+
+# Dictionary of fade requests
+var _faders : Dictionary = {}
+
+# Fade update flag
+var _update : bool = false
+
+# Fade mesh
+var _mesh : MeshInstance3D
+
+# Fade shader material
+var _material : ShaderMaterial
+
+
+# Add support for is_xr_class on XRTools classes
+func is_xr_class(name : String) -> bool:
+	return name == "XRToolsFade"
+
+
+# Called when the fade node is ready
+func _ready() -> void:
+	# Add to the fade_mesh group - in the future this should be replaced with
+	# static instances.
+	add_to_group("fade_mesh")
+
+	# Get the mesh and material
+	_mesh = $FadeMesh
+	_material = _mesh.get_surface_override_material(0)
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(_delta : float) -> void:
+	# Skip if nothing to update
+	if not _update:
+		return
+
+	# Calculate the cumulative shade color
+	var fade := Color(1, 1, 1, 0)
+	var show := false
+	for whom in _faders:
+		var color := _faders[whom] as Color
+		fade = fade.blend(color)
+		show = true
+
+	# Set the shader and show if necessary
+	_material.set_shader_parameter("albedo", fade)
+	_mesh.visible = show
```

Comment:
You'll want to add an extra check here if alpha = 0. If the fade is invisible but you don't make the node invisible, you're drawing a full screen rect that is blending with a fully transparent colour. That is unnecessary overhead.

Reply:
Never mind, just noticed the alpha check below :P